clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/schema/HDFSRmdSchemaSource.java

```diff
@@ -0,0 +1,138 @@
+package com.linkedin.venice.hadoop.schema;
+
+import com.linkedin.venice.annotation.NotThreadsafe;
+import com.linkedin.venice.controllerapi.ControllerClient;
+import com.linkedin.venice.controllerapi.MultiSchemaResponse;
+import com.linkedin.venice.schema.rmd.RmdVersionId;
+import com.linkedin.venice.utils.Utils;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.avro.Schema;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * HDFSRmdSchemaSource caches the RMD schemes for a given store on the HDFS and serve them during the Mapper stage
+ * It will make API calls on Venice controller to retrieve all RMD schemes.
+ * <p>
+ * Borrowed some codes from HDFSSchemaSource. The difference is HDFSRmdSchemaSource is created for a single store.
+ */
+@NotThreadsafe
+public class HDFSRmdSchemaSource implements RmdSchemaSource, AutoCloseable {
+  private static final Logger LOGGER = LogManager.getLogger(HDFSRmdSchemaSource.class);
+  private static final String UNDERSCORE = "_";
+  private static final String SEPARATOR = "/";
+
+  private final String storeName;
+  private final FileSystem fs;
+  private final Path schemaDir;
+  private final ControllerClient controllerClient;
+
+  public HDFSRmdSchemaSource(final String schemaDir, final String storeName, final ControllerClient controllerClient)
+      throws IOException {
+    Configuration conf = new Configuration();
+    this.fs = FileSystem.get(conf);
+    this.schemaDir = new Path(schemaDir);
+    if (!fs.exists(this.schemaDir)) {
+      fs.mkdirs(this.schemaDir);
+    }
+    this.storeName = storeName;
+    this.controllerClient = controllerClient;
```

Comment:
> Shall we pass a ControllerClient instance to loadRmdSchemasOnDisk

I think that's a good idea, also making the test more easy too!

>  it is good to make loadRmdSchemasOnDisk synchronized to avoid concurrent load and maintain a flag

I thought about that but think `HDFSRmdSchemaSource` is not intended to run in a multi-threading env, in other words, it's just like an Azkaban job, one instance per push job. Also, the load happens on Azkaban job whereas the fetch happens on mapper stage so no flag is needed atm. For the time being, I've added `NotThreadsafe` annotation on this class so people is aware of this. Thanks for this suggestion!